Kernel/Coredump.cpp

```diff
@@ -37,18 +51,27 @@ ErrorOr<NonnullOwnPtr<Coredump>> Coredump::try_create(NonnullLockRefPtr<Process>
         return EPERM;
     }
 
+    Vector<FlatRegionData> regions;
+    {
+        SpinlockLocker lock(process->address_space().get_lock());
+        regions.ensure_capacity(process->address_space().regions().size());
```

Comment:
Allocating memory under the address space spinlock seems like a dangerous game? :smile: 

I guess the process is suspended at this point though? So there should be no other threads active?

Reply:
This allocates using the kernel address space though, not the process address space that is locked, right? It _might_ be an issue regarding potentially doing something heavy while holding a spinlock, but nothing more than that. And I don't see how it can be avoided either if we absolutely want to make sure that our data stays consistent.

The process should be suspended though, assuming that we are currently coredumping after an assert failed or something. Do we even have any other code paths where coredumps are created?

Reply:
Actually, I guess I can just move it outside since this is just about the capacity anyways. Even if it somehow gets inconsistent, it should not have a significant impact because it will just continue to allocate as needed.